batchflow/models/tf/layers/utils.py

```diff
@@ -0,0 +1,22 @@
+""" Common utility functions for all of TensorFlow layers. """
+import inspect
+
+
+
+def add_as_function(cls):
+    """ Decorator for classes. Automatically adds functional interface for `call` method of class. """
+    name = cls.__name__
+    func_name = ''.join('_' + c.lower()
+                        if (c.isupper() and (i != len(name)-1) and name[i+1].islower()) else c.lower()
```

Comment:
c.lower() if ... else c.lower() ???

Reply:
Join  `'_' + c.lower()` or `c.lower()`

Reply:
`''.join(c if c.islower() else ('_' + c.lower()) for c in name).lstrip('_')`